src/EditorFeatures/CSharp/BlockCommentCompletion/BlockCommentCompletionCommandHandler.cs

```diff
@@ -0,0 +1,171 @@
+            }
+
+            var currentLineStartsWithBlockCommentStartString = snapshotLine.StartsWith(firstNonWhitespacePosition, "/*", false);
+            var currentLineStartsWithBlockCommentEndString = snapshotLine.StartsWith(firstNonWhitespacePosition, "*/", false);
+            var currentLineStartsWithBlockCommentMiddleString = snapshotLine.StartsWith(firstNonWhitespacePosition, "*", false);
+
+            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            {
+                return null;
+            }
+
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
+            {
+                return null;
+            }
+
+            if (currentLineStartsWithBlockCommentStartString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*|*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      /|*
+                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                }
+                else
+                {
+                    //      /*|
+                    return " *" + GetPaddingOrIndentation(snapshotLine, caretPosition, firstNonWhitespacePosition, "/*");
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentEndString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*
+                    //      |*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      *|/
+                    return "*";
+                }
+                else
+                {
+                    //      /*
+                    //   |   */
+                    return " * ";
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentMiddleString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      *|*/
+                    return "";
+                }
+                else if (caretPosition > firstNonWhitespacePosition)
+                {
+                    //      *|
+                    return "*" + GetPaddingOrIndentation(snapshotLine, caretPosition, firstNonWhitespacePosition, "*");
+                }
+                else
+                {
+                    //      /*
+                    //   |   *
+                    return " * ";
+                }
+            }
+
+            return null;
+        }
+
+        private static bool BlockCommentEndsRightAfterCaret(SnapshotPoint caretPosition) => caretPosition.Snapshot.GetText(caretPosition, 2) == "*/";
```

Comment:
Almost certainly not.  Looks like just an off by 1 error that never bit us before.

@dpoeschl  I told you counting was just about the most difficult thing in computing!